lib/prototypes.h

```diff
@@ -44,6 +44,12 @@ extern int add_groups (const char *);
 /* age.c */
 extern void agecheck (/*@null@*/const struct spwd *);
 extern int expire (const struct passwd *, /*@null@*/const struct spwd *);
+
+/* agetpass.c */
+extern void erase_pass(char *p);
+__attribute__((malloc(erase_pass)))
```

Comment:
This function annotation does not work with Clang 15:
```
In file included from basename.c:21:
../lib/prototypes.h:50:16: error: 'malloc' attribute takes no arguments
__attribute__((malloc(erase_pass)))
               ^
1 error generated.
```

Reply:
Confirmed:

```c
alx@asus5775:~/tmp$ cat malloc.c 
#include <stdlib.h>


void my_free(void *p);
[[gnu::malloc(my_free)]] void *my_malloc(size_t s);


void my_free(void *p)
{
	free(p);
}

void *my_malloc(size_t s)
{
	return malloc(s);
}
```
```c
alx@asus5775:~/tmp$ gcc -Wall -Wextra -S malloc.c 
```
```c
alx@asus5775:~/tmp$ clang -Wall -Wextra -S malloc.c 
malloc.c:5:2: error: expected expression
[[gnu::malloc(my_free)]] void *my_malloc(size_t s);
 ^
malloc.c:5:26: error: expected identifier or '('
[[gnu::malloc(my_free)]] void *my_malloc(size_t s);
                         ^
2 errors generated.
```
We'll need to hide it in an `ATTR_MALLOC` macro.

Reply:
BTW, the fact that this results in an error is due to a GCC bogus design decision.  They called to completely different attributes with the same name.

If they called it differently, one should be able to use `[[]]` syntax and force the compiler to ignore unknown attributes.